src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,52 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        final LocalDate dateStart = LocalDate.parse(dateFrom, formatter);
+        final LocalDate dateFinish = LocalDate.parse(dateTo, formatter);
+        final int namesNumber = names.length;
+        LocalDate dateEmployee;
+        StringBuilder[] results = new StringBuilder[namesNumber + 1];
+        int[] salary = new int[namesNumber];
+
+        results[0] = new StringBuilder("Report for period ");
+        results[0].append(dateFrom).append(" - ").append(dateTo);
+
+        if (data == null) {
+            return "";
+        }
```

Comment:
Try to implement without an array
```suggestion
     StringBuilder report = new StringBuilder("Report for period " + dateFrom + " - " + dateTo);
```